src/main/java/com/sivalabs/springblog/comment/domain/CommentEntity.java

```diff
@@ -0,0 +1,105 @@
+package com.sivalabs.springblog.comment.domain;
+
+import com.sivalabs.springblog.post.domain.PostEntity;
+import com.sivalabs.springblog.user.domain.UserEntity;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.SequenceGenerator;
+import jakarta.persistence.Table;
+import java.time.LocalDateTime;
+
+@Entity
+@Table(name = "comments")
+public class CommentEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "comment_id_seq_gen")
+    @SequenceGenerator(name = "comment_id_seq_gen", sequenceName = "comment_id_seq", allocationSize = 50)
+    private Long id;
+
+    @Column(nullable = false, length = 100)
+    private String name;
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    private String content;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "post_id", nullable = false)
+    private PostEntity post;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "created_by", nullable = false)
+    private UserEntity createdBy;
+
+    @Column(name = "created_date", nullable = false, updatable = false)
+    private LocalDateTime createdDate;
+
+    // Default constructor
+    public CommentEntity() {}
+
+    // Constructor with fields
+    public CommentEntity(
+            Long id, String name, String content, PostEntity post, UserEntity createdBy, LocalDateTime createdDate) {
+        this.id = id;
+        this.name = name;
+        this.content = content;
+        this.post = post;
+        this.createdBy = createdBy;
+        this.createdDate = createdDate;
+    }
+
+    // Getters and Setters
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
```

Comment:
Try to use fluent setters so that it becomes easier when writing Junits or bigger entity 


all setters can be replaced with  as below pattern

```java
public CommentEntity setName(String name) {
        this.name = name;
        return this;
}
```

Reply:
Thank you for the suggestion! While I understand the benefits of fluent setters, in this particular case, I don't see a strong need to change. I'm using `Testcontainers` for testing, and the current approach aligns well with our practices. Additionally, standard setters are simple and work seamlessly with JPA and other tools. I'll keep this in mind if a clear use case for fluent setters arises in the future.